docs/resources/git_setup.md

```diff
@@ -2,6 +2,119 @@
 
 How to's concerning  `git` in general,  [`gitlab`][cern_gitlab], [`github`][github] and CI.
 
+## Github Commandline Access Quickstart
+
+This section explains the basic steps to get started with `github`.
+Since a few years, github has disabled access via password only for security reasons,
+so you need to cre
+This aims to be as short and concise as possible, for more extenive information, [see the github security documentation][github_security].
+
+### Setup SSH Access
+
+An easy way to access github securely is to use SSH.
+
+#### Create SSH Key
+
+For this, you first need to create a SSH key pair on your computer using the email address of your github account.
+
+```bash
+ssh-keygen -t ed25519 -C "your_email@example.com"
+```
+
+When asked for a location, it make sense to give it an easily identifiable name, so you will know what the key is for.
+The file should be placed in `~/.ssh/`, unless you are on `afs`, in which case the `~/private/` directory should be used.
+
+```text
+~/.ssh/github_authenticate
+```
+
+!!! tip "Passphrase"
+    You can optionally provide a passphrase for the key, which will make it more secure.
+    This way, even if someone else gets a hold of the private key file, they will not be able to access it.
+    On the downside, you will be asked to enter the passphrase every time you want to use the key.
+    **It recommended, to use a passphrase** but as it is just an extra layer of security, you can keep it short and simple.
+
+This will create two files: `github_authenticate` and `github_authenticate.pub`.
+The `.pub` is your public key that you can share with others,
+while the other file is your private key and **should never be shared with anyone!**
+
+!!! quote "Keep it secret, keep it safe!"
+    _Gandalf_, about private SSH keys (probably).
+
+#### Add SSH Key to Github
+
+After creating the key, you need to add it to your github account.
+For this you need to log into your github account, click on your avatar and go to `Settings` &rarr; `SSH and GPG keys`.
+Then click on [++"New SSH key"++{.green-gui-button}][github_new_ssh_key]{target=_blank} and paste the contents of the `.pub` file into the `Key` field.
+
+Give it a resonable name in the `Title` field and leave the `Key type` as `Authentication key`.
```

Comment:
```suggestion
Give it a resonable name in the `Title` field (which it will appear as in the GitHub interface) and leave the `Key type` as `Authentication key`.
```